pyproject.toml

```diff
@@ -61,9 +61,8 @@ test = [
 # release scripts and process.
 dev-release = [
     "towncrier ~= 23.6",
-    "pydoctor ~= 23.4.0",
-    "sphinx-rtd-theme ~= 1.2",
-    "readthedocs-sphinx-ext ~= 2.2",
```

Comment:
I am not sure why the sphinx-ext was pinned here.
We only need the `sphinx-rtd-theme` and it should pull the dependencies

Reply:
why are we pinning *anything* here? we should have a requirements.txt with exact pins in it for CI; these should be pretty loose.

Reply:
These are used for both CI and local dev.

I think it's important to pin the direct dependencies.

Also, I prefer to have all the dependendencies in the same pyproject.toml file, rather than having requirements.txt and then other deps inside readthedocs.yaml or GitHub Actions yaml

Reply:
> These are used for both CI and local dev.
> 
> I think it's important to pin the direct dependencies.
> 
> Also, I prefer to have all the dependendencies in the same pyproject.toml file, rather than having requirements.txt and then other deps inside readthedocs.yaml or GitHub Actions yaml

I think that we should be using `pip-compile` to produce the pinned dependencies *from* the `pyproject.toml`, not manually duplicating them :)

Reply:
I am not sure I got this.
These changes are done in `pyproject.toml`
Also, with the latest changes, we should now have only top-level dependencies declared.

-------

I have never used `pip-compile` . I am not against it.
But so far, I think that we are ok, and we other other issues, bigger than pinning the deps for the dev tools.